src/keyboard_drag_strategy.ts

```diff
@@ -35,7 +36,24 @@ export class KeyboardDragStrategy extends dragging.BlockDragStrategy {
   /** Where a constrained movement should start when traversing the tree. */
   private searchNode: ASTNode | null = null;
 
+  private cursor: LineCursor | null;
+
+  isNewBlock: boolean;
+
+  constructor(
+    private block: BlockSvg,
+    private insertStartPoint: RenderedConnection | null,
+  ) {
+    super(block);
+    this.isNewBlock = !!this.insertStartPoint;
+    this.cursor = block.workspace.getCursor();
+  }
+
   override startDrag(e?: PointerEvent) {
+    if (!this.cursor) throw new Error('precondition failure');
+    // Select and focus block.
+    common.setSelected(this.block);
```

Comment:
Won't `setCurNode` guarantee selection? It'd be nice to consolidate since otherwise every setSelected() needs to be reconsidered for focus.